kernel/modules: make sure igb loads at boot

riptidewave93 opened the pull request:
This is to make sure that AutoLoad adds igb to modules-boot.d
so it's accessible via initramfs during boot. This is to ensure that
networking works when in recovery mode, and are loaded before
etc/board.d/02_network is called. Note that other network drivers
already have this set, such as tg3.

Signed-off-by: Chris Blake <chrisrblake93@gmail.com>

riptidewave93 commented:
CC @chunkeey

lynxis commented:
Hi @riptidewave93 ,
the code looks good. The commit title is great but the commit message is a little bit indirect.
E.g. "Without loading the igb at boot, the recovery wouldn't have network available. All network drivers should be loaded before etc/board.d/02_network is called."
(As reference I've learned a lot from http://who-t.blogspot.de/2009/12/on-commit-messages.html )

riptidewave93 commented:
Thanks for the feedback @lynxis I'll work on rewording it.

riptidewave93 commented:
@lynxis fixed and awaiting review

adschm commented:
One could add a
`Fixes: 7e0e5110bc90 ("kernel: add igb kernel module")`
if I understand the issue correctly.

riptidewave93 committed:
kernel/modules: make sure igb loads at boot

Without loading the igb at boot, the recovery wouldn't have
network available. All network drivers should be loaded before
etc/board.d/02_network is called. Note that other network drivers
already have this set, such as tg3.

Fixes: 7e0e5110bc90 ("kernel: add igb kernel module")
Signed-off-by: Chris Blake <chrisrblake93@gmail.com>

riptidewave93 commented:
@adschm updated, thanks

aparcar commented:
Merged, thanks!

remote: To github.com:openwrt/openwrt.git
remote:    92ac2a20eb..f1e41155c9  master -> master

zyphlar commented:
How would the igb driver be loaded otherwise? This fix doesn't seem to appear in v21.02.0 and I'm trying to figure out why my build doesn't load the driver. Is there some init script elsewhere that polls for hardware and loads modules dynamically?

riptidewave93 commented:
@zyphlar Without this setting, instead of being loaded in the initramfs it would be loaded after the filesystem is mounted. If you are having issues loading igb, this is probably not the fix for your problem since it should still load once the filesystem mounts.

zyphlar commented:
@riptidewave93 thanks, I seem to be missing whatever other section that is, do you know what that post-filesystem stuff might be?

riptidewave93 commented:
@zyphlar I recommend you look at the init scripts in https://github.com/openwrt/openwrt/tree/master/package/base-files/files/lib/preinit, which are referenced by https://github.com/openwrt/openwrt/blob/master/package/base-files/files/etc/preinit
